src/Kestrel.Core/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -74,10 +74,11 @@ public void Dispose()
             }
         }
 
-        // Review: This is called when a CancellationToken fires mid-write. In HTTP/1.x, this aborts the entire connection.
-        // Should we do that here?
+        // This is called when a CancellationToken fires mid-write. In HTTP/1.x, this aborts the entire connection.
+        // For HTTP/2 we abort the stream.
         void IHttpOutputAborter.Abort(ConnectionAbortedException abortReason)
         {
+            _stream.ResetAndAbort(abortReason, Http2ErrorCode.INTERNAL_ERROR);
```

Comment:
You add a stream back reference for trailers, and all of the sudden we have tight coupling! 😆

Reply:
But so convenient!